src/render/PointArray.cpp

```diff
@@ -386,14 +386,83 @@ bool PointArray::loadFile(QString fileName, size_t maxPointCount)
     {
         g_logger.debug("Reordering field %d: %s", i, m_fields[i]);
         reorder(m_fields[i], inds.get(), m_npoints);
-        emit loadProgress(int(100*(i+1)/m_fields.size()));
+        emit loadProgress(int(100*(i+1)/(m_fields.size()+1))); // denominator +1 for permutation reorder below
     }
     m_P = (V3f*)m_fields[m_positionFieldIdx].as<float>();
+    // The index we want to store is the reverse permutation of the index above
+    m_inds = std::unique_ptr<size_t[]>(new size_t[m_npoints]);
+    for (size_t i = 0; i < m_npoints; ++i) m_inds[inds[i]] = i;
+    emit loadProgress(int(100));
 
     return true;
 }
 
 
+void PointArray::mutate(std::shared_ptr<GeometryMutator> mutator)
+{
+    // Now we need to find the matching columns
+    auto npoints = mutator->pointCount();
+    const std::vector<GeomField>& mut_fields = mutator->fields();
+    auto mut_idx = mutator->index();
+
+    if (m_npoints > 4294967296)
```

Comment:
Just in case it changes one day? ;)

Will do. The thing about magic macros is that you have to know they exist. :)